docs/examples/java-examples.md

```diff
@@ -0,0 +1,240 @@
+
+```kotlin
+  fun getUserDsnpKeyBundles(long msaId, int schemaId) {
+    // Retrieve current public keys from the chain.
+    val publishedKeys = frequencyClient.getItemizedStorage(msaId, publicKeySchemaId).join()
+
+    // ----- DESERIALIZATION OF PUBLISHED ON-CHAIN KEYS ----- //
+    // Deserialize Published Public Graph Keys to raw form
+    val mappedKeys = publishedKeys.items.map { i ->
+      KeyData.newBuilder()
+        .setIndex(i.index)
+        .setContent(
+          // substring use is to skip the `0x` in front of the payload
+          ByteString.fromHex(i.payload.substring(2)))
+        .build()
+    }
+    val dsnpKeys = DsnpKeys.newBuilder()
+      .setDsnpUserId(graphOwnerMsaId.toLong())
+      .setKeysHash(publishedKeys.contentHash)
+      .addAllKeys(mappedKeys)
+      .build()
+
+    return ImportBundles.newBuilder()
+      .addBundles(
+        ImportBundles.ImportBundle.newBuilder()
+          .setDsnpUserId(msaId)
+          .setSchemaId(schemaId)
+          .setDsnpKeys(dsnpKeys)
+          .build()
+      ).build()
+  }
+
+  fun getUserGraphKeyPairBundle(
+    long msaId,
+    List<GraphKeyPair> currentKeyPairs {
+
+    return ImportBundles.newBuilder()
+      .addBundles(
+        ImportBundles.ImportBundle.newBuilder()
+          .setDsnpUserId(msaId)
+          .setSchemaId(schemaId)
+          .addAllKeypairs(currentKeyPairs)
+          .build()
+      ).build()
+}
+
+fun addNewUserGraphKeypair(long msaId, int schemaId, GraphKeyPair newKeypair)
+{
+    val bundles = getUserDsnpKeyBundles(msaId, schemaId));
+    graph.importUserData(bundles);
+
+    val graphActions = Actions.newBuilder().addActions(
+      Actions.Action.newBuilder().setAddKeyAction(
+        Actions.Action.AddGraphKey.newBuilder()
+          .setOwnerDsnpUserId(msaId)
+          .setNewPublicKey(newKeyPair.publicKey)
+          .build()
+      )
+    ).build()
+
+    // add new key to graph
+    graph.applyActions(graphActions)
+
+    // export graph updates which in this case is only the added new key
+    val updates = graph.exportUpdates()
+
+    // ----- PUBLISH TO BLOCKCHAIN ----- //
+    // map exported key into chain request
+    val itemizedActions: List<ItemAction> = updates.filter { u -> u.hasAddKey() }.map { update ->
+      AddItemAction.from(update.addKey.payload.toByteArray())
+    }
+
+    // finally, add the new exported key to blockchain
+    // here we are using `createApplyItemActions` for simplicity but in reality this should be done using
+    // `createApplyItemActionsWithSignature` since user should sign the payload of adding a new graph key with one of
+    // their msa control keys
+    val retVal =
+      frequencyClient.createApplyItemActions(graphOwnerMsaId, publicKeySchemaId, keyContentHash.toBigInteger(), itemizedActions)
+        .join()
```

Comment:
```suggestion
    // finally, add the new exported key to blockchain
    // the user MUST sign the payload of adding a new graph key with one of their msa control keys
    // The full data to sign is available: https://libertydsnp.github.io/frequency/pallet_stateful_storage/types/struct.ItemizedSignaturePayload.html
    val retVal =
      frequencyClient.createApplyItemActionsWithSignature(graphOwnerSigningPublicKey, signatureProof, publicKeySchemaId, keyContentHash.toBigInteger(), itemizedActions)
        .join()
```

Reply:
Ah, good catch--I forgot the GraphPublicKey schema was `SignatureRequired` in addition to `AppendOnly`